core/src/main/scala/org/apache/spark/status/api/v1/StagesResource.scala

```diff
@@ -40,8 +43,27 @@ private[v1] class StagesResource extends BaseAppResource {
       @PathParam("stageId") stageId: Int,
       @QueryParam("details") @DefaultValue("true") details: Boolean): Seq[StageData] = {
     withUI { ui =>
-      val ret = ui.store.stageData(stageId, details = details)
+      var ret = ui.store.stageData(stageId, details = details)
       if (ret.nonEmpty) {
```

Comment:
going with my comment above, I don't want to be modifying the StageData by changing the parameters to var, we should either get all the data we need from stageData or we could perhaps create a separate call to get these.  Will have to look in more depth

Reply:
Done